next/backend/utils/forms.ts

```diff
@@ -209,19 +171,42 @@ export const removeNeedlessXmlTransformArraysRecursive = (
   return obj
 }
 
-// TODO create ajv instance once for BE, add async validations
-export const validateDataWithJsonSchema = (data: any, schema: any) => {
+const checkIsPhone = (schema: any, value: string) => {
+  // TODD: verify user in db
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(false), 500)
+  })
+}
+
+export const validateDataWithJsonSchema = async (data: any, schema: any) => {
   const ajv = new Ajv()
   addFormats(ajv)
   ajv.addFormat('data-url', () => true)
   ajv.addFormat('ciselnik', () => true)
 
   ajv.addKeyword('example')
   ajv.addKeyword('enumNames')
+  ajv.addKeyword('isToken')
+  ajv.addKeyword('disabled')
+  ajv.addKeyword('tooltip')
+  ajv.addKeyword('error')
+  ajv.addKeyword({
+    keyword: 'isPhone',
```

Comment:
for this not to confuse, as we will not use that endpoint for real phone otp and I can see this being strange to someone opening the project later down the line without context 

Let's rename everything used in this example async validation so that it is clear it's just an example - ie. keyword `isExampleAsyncValidation`, the validate function `exampleAsyncValidation`, endpoint etc etc. Where warranted (i.e. the api endpoint) add additional comments

make it so that if the field validated this way is empty the validation fails, otherwise it passes

furthermore, quite a tricky part of this setup is the question of calling the same validation from FE and from BE - it likely won't be the exactly same call (as FE will need a network call and BE won't), but it shouldn't be two completely different functions either. Find some nice way of how to present and organise this.